mirgecom/initializers.py

```diff
@@ -54,60 +54,62 @@
-    if gas_model is None:
-        raise ValueError("Gas model is required to create a FluidState.")
+    if gas_model is None and eos is None:
+        raise ValueError("Must provide 1 of (gas_model, eos).")
+    if eos is None:
+        eos = gas_model.eos
+    if return_fluid_state and gas_model is None:
+        raise ValueError("Must provide gas_model to create a fluid state.")
 
     state_spec = [pressure is None, temperature is None, density is None]
     if sum(state_spec) != 1:
         raise ValueError("Must provide 2 of (pressure, temperature, density).")
 
-    dim = dcoll.dim
+    dim = nodes.shape[0]
 
-    if dd_bdry is None:
-        nodes = actx.thaw(dcoll.nodes())
-    else:
-        data_discr = dcoll.discr_from_dd(dd_bdry)
-        nodes = actx.thaw(data_discr.nodes())
     zeros = nodes[0]*0.0
 
     if velocity is None:
-        velocity = np.zeros(dim)
+        velocity = np.zeros(dim,)
 
     if pressure is None:
-        pressure = gas_model.eos.get_pressure(density, temperature, mass_fractions)
+        pressure = eos.get_pressure(density, temperature, species_mass_fractions)
 
     if temperature is None:
-        temperature = gas_model.eos.get_temperature(density, pressure,
-                                                    mass_fractions)
+        gas_const = eos.gas_const(species_mass_fractions=species_mass_fractions)
+        temperature = pressure/(density*gas_const)
 
     if density is None:
-        density = gas_model.eos.get_density(pressure, temperature, mass_fractions)
-
-    internal_energy = gas_model.eos.get_internal_energy(temperature, mass_fractions)
+        density = eos.get_density(pressure, temperature, species_mass_fractions)
 
-    species_mass = None if mass_fractions is None \
-        else (density + zeros)*mass_fractions
-
-    total_energy = density*internal_energy + density*np.dot(velocity, velocity)/2
     momentum = density*velocity
+    energy = density*(eos.get_internal_energy(temperature, species_mass_fractions)
+                      + 0.5*np.dot(velocity, velocity))
+
+    if species_mass_fractions is None:
+        species_mass = None
+    else:
+        nspecies = len(species_mass_fractions)
+        species_mass = make_obj_array([density*species_mass_fractions[i] + zeros
+                                       for i in range(nspecies)])
 
     cv = make_conserved(dim=dim, mass=density + zeros,
-                        energy=total_energy + zeros,
+                        energy=energy + zeros,
                         momentum=momentum + zeros,
                         species_mass=species_mass)
 
-    if make_fluid_state:
-        from mirgecom.gas_model import make_fluid_state
+    if return_fluid_state:
         return make_fluid_state(cv=cv, gas_model=gas_model,
-            temperature_seed=temperature)
-    else:
-        return cv
+                                temperature_seed=temperature)
+
+    return cv
```

Comment:
Is there a good reason to make this do the `make_fluid_state` part?   I wonder whether this function should always just return CV and let the user make the state if they want?  Then maybe we don't need to do the song-and-dance with the EOS vs. gas_model.

Reply:
Since all functions in `initializers` return cv, I agree. Removed the fluid_state option.